frontend/rolecall/src/app/cast/cast-drag-and-drop.component.ts

```diff
@@ -127,206 +153,150 @@ export class CastDragAndDrop implements OnInit {
         }
       })
     };
     return newCast;
   }
-
-  /** Gets the max number of dancers in any specific position (i.e. max number
-   * of subcasts [the columns]) for the position in the cast */
-  getMaxNumberInDancerPositionForCast(positionIndex: number): number {
-    let numSubCasts = 0;
-    this.cast.filled_positions[positionIndex].groups.forEach((val, ind) => {
-      if (val.group_index + 1 > numSubCasts) {
-        numSubCasts = val.group_index + 1;
-      }
-    });
-    return numSubCasts;
-  }
-
-  /** Gets the largest number of individual dancers in a position (the max rows)*/
-  getMaxDancerIndexForCast(positionIndex: number): number {
-    let numBackups = 0;
-    this.cast.filled_positions[positionIndex].groups.forEach((val, ind) => {
-      if (val.members.length > numBackups) {
-        numBackups = val.members.length;
-      }
-    });
-    return numBackups;
-  }
-
-  /** Gets the max number of dancers in any specific position (i.e. max number
-   * of subcasts [the columns]) for the position in the drag and drop data */
-  getMaxNumberInDancerPositionForData(positionIndex: number): number {
-    let numSubCasts = 0;
-    if (isNullOrUndefined(this.data[positionIndex])) {
-      return 0;
-    }
-    this.data[positionIndex].forEach(val => {
-      if (val.length > numSubCasts) {
-        numSubCasts = val.length;
-      }
-    });
-    return numSubCasts;
-  }
-
-  setColumnHeaders(positionIndex: number) {
-    this.columnHeaders[positionIndex] = [];
-    let pipe = new NumberToPlacePipe();
-    this.columnHeaders[positionIndex] = Array(this.getMaxNumberInDancerPositionForData(positionIndex) + 1).fill(0).map((x, i) => pipe.transform(i + 1) + " Cast");
-  }
-
-  ensureEmptyArrayAtEnd(positionIndex: number) {
-    this.data[positionIndex] = this.data[positionIndex].filter(row => row.length != 0);
-    this.data[positionIndex].push([]);
-  }
-
-  updateEmptyRows(positionIndex: number) {
-    let posData = this.data[positionIndex];
-    this.emptyCells[positionIndex] = Array(posData.length);
-    let numHeaders = this.columnHeaders[positionIndex].length;
-    for (let i = 0; i < posData.length; i++) {
-      this.emptyCells[positionIndex][i] = Array(numHeaders - posData[i].length).map((x, ind) => "Empty Cell " + (ind + 1));
-    }
-  }
-
-  ensureAllPositionsMet(cast: Cast) {
-    this.positionVals = this.pieceAPI.pieces.get(this.cast.segment).positions;
-    for (let pos of this.positionVals) {
-      if (!cast.filled_positions.find(val => val.position_uuid == pos.uuid)) {
-        cast.filled_positions.push({
-          position_uuid: pos.uuid,
-          groups: [
-            {
-              group_index: 0,
-              members: []
-            }
-          ]
-        });
-      }
-    }
-  }
-
+  
   setupData() {
     if (!this.castSelected) {
       return;
     }
     if (!this.castAPI.hasCast(this.selectedCastUUID)) {
-      this.data = [[]];
+      //this.data = [[]];
+      this.castPositions = [];
```

Comment:
Can move this line up out of the if-statement, it's repeated in both the 'if' and the 'else' cases

Reply:
Done.